src/ChatMessageBuilder.h

```diff
@@ -0,0 +1,107 @@
+
+// ChatMessageBuilder.h
+
+// Declares the cChatMessageBuilder class used to build a chat message
+
+#include "Defines.h"
+#include "json/json.h"
+#include "Logger.h"
+#include "ChatColor.h"
+
+// tolua_begin
+class cChatMessageBuilder
+{
+public:
+	// tolua_end
+
+	/** This class is used to store the text, the click event, hover event and insertion text. */
+	class cChatMessagePart
+	{
+	public:
+		cChatMessagePart()
+		{
+		}
+
+		cChatMessagePart(const AString & a_Text):
+			m_Text(a_Text)
+		{
+		}
+
+
+		~cChatMessagePart() {}
+
+		AString m_Text;
+		AString m_InsertionText;
+		AString m_HoverText;
+		AString m_ClickText;
+		AString m_HoverAction;
+		AString m_ClickAction;
+	} ;
+
+	typedef std::unique_ptr<cChatMessagePart> cChatMessagePartPtr;
+	typedef std::vector<cChatMessagePartPtr> cParts;
+
+	cParts m_Parts;
```

Comment:
This should probably be `protected`, too. And you should add a note that it is never empty, so that none of the modifiers need to check.